AirCasting/SearchAndFollow/AirBeamMeasurements/AirBeamMeasurementsDownloader.swift

```diff
@@ -31,32 +32,31 @@ final class AirBeamMeasurementsDownloaderDefault: AirBeamMeasurementsDownloader
                         combinedResult.append(resultPart)
                         group.leave()
                     } catch {
+                        allErrors.append(error)
                         Log.info("Error when downloading one of the AirBeam streams: \(error)")
                         group.leave()
-                        completion(.failure(error))
                     }
                 case .failure(let error):
+                    allErrors.append(error)
                     Log.info("Error when downloading one of the AirBeam streams: \(error)")
                     group.leave()
-                    completion(.failure(error))
                 }
             }
         }
         group.notify(queue: .main) {
+            if !allErrors.isEmpty { completion(.failure(allErrors.first!)) }
```

Comment:
we're only passing the first error here. I think that's reasonable here, but the *perfect* solution would be to pass all under a [composite](https://refactoring.guru/design-patterns/composite) error. Could look like this:
```Swift
struct CompositeError: Error, LocalizedError {
    let underlyingErrors: [Error]
    public var errorDescription: String? {
        // nice description that prints out all underlying descriptions
    }
}
```